api/grpcserver/transaction_service.go

```diff
@@ -20,6 +21,14 @@ type TransactionService struct {
 	Network api.NetworkAPI // P2P Swarm
 	Mesh    api.TxAPI      // Mesh
 	Mempool *state.TxMempool
+	syncer  Syncer
+}
+
+// Syncer defines ways of interacting with the node syncer
+// business logic
+type Syncer interface {
```

Comment:
Please use the existing interface: https://github.com/spacemeshos/go-spacemesh/blob/fbac657c8e420a4cfea4f82dbc9eb748c348a211/api/node.go#L60

Reply:
Why use this interface and implement Start as no-op? It doesn't make sense. I know it's good to reutilize the code but this basically states, "hey I do want  Syncer and that Syncer can decide if it's synced or not and I don't require anything else".

Reply:
I'm not sure what you mean. You don't need to implement Start, or IsSynced. The `app.syncer` object that you pass in at https://github.com/spacemeshos/go-spacemesh/pull/2114/files#diff-a41cea38155fea30556048c9d149900fR742 already implements both of these.

Reply:
I'm talking about  "interface-segregation principle". (no client should be forced to depend on methods it does not use). Given this argument, if you see the `TransactionService` shouldn't make clients implement the whole `Syncer` interface if the `TransactionService` dosen't require it.